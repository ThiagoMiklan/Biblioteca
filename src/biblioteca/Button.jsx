// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import {renderToString} from 'react-dom/server';

type Props = {
  disabled?: bool,
  delete?: bool,
  definition?: string,
  children?: React.Node,
  custom?: any,
  onClick?: () => void
}

/*

Possui children
Eventos: onClick no <button> </button>
Observações: 

    1 - Caso exista a necessidade de adicionar um ícone ao componente Button, é necessário seguir
             a seguinte estrutura:

            <Button definition="info" >
                <Icon icon_name="fas fa-check" [right={true} OR empty prop right]>
                    Nome do botão
                </Icon>
            </Button >

            Sendo necessário informar o componente Icon como filho do componente Button
            e o nome do botão deve ser informado no Icon e não no Button como em outras
            situações onde usa-se o componente isolado.
*/

const Button= (props: Props)=>{
    var definition = getDefinition(props);
    
     return <button 
                {...props.custom} 
                disabled = {props.disabled == true? true: false } 
                className={definition} 
                onClick={props.onClick}
                >
                {props.children}
         </button>
         ;
}

function getDefinition(props){
    var definition = "";

    if(props.delete ==  true){
        definition = "delete";
    }else{
        definition = (props.definition == undefined) ? "button" : "button "+ props.definition;
    }

    return definition;
}

Button.propTypes = {
    definition: PropTypes.string,
    onClick: PropTypes.func,
    delete: PropTypes.bool,
    disabled: PropTypes.bool
};


export default Button;