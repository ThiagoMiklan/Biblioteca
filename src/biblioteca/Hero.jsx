// @flow
import React from 'react';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import PropTypes from 'prop-types';
import checkerDefinition from '../tools/checker.js';

// v.1.0

type Props = {
    title?: string,
    subtitle?: string,
    definition?: string
}

const Hero = (props: Props)=>{
  checkerDefinition(props.definition,"Hero");
  var hero_definition =  (props.definition ==  undefined) ? "": props.definition;
  var definition =  classnames(getClassName("hero "+ hero_definition,"Hero"));
    return (
    <section className={definition}>
    <div className="hero-body">
      <div className="container">
        <h1 className="title">
         {props.title}
        </h1>
        <h2 className="subtitle">
          {props.subtitle}
        </h2>
      </div>
    </div>
  </section>);
}

Hero.propTypes ={
    // Título do Banner
    title: PropTypes.string,
    // Subtítulo do Banner
    subtitle: PropTypes.string,
    // Características do Bulma
    definition: PropTypes.string
}

export default Hero;
