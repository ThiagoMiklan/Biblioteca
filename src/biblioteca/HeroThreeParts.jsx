import React from 'react';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import PropTypes from 'prop-types';
import {InvalidNumberChilds,assembleDefaultMessage} from '../exception/InvalidNumberChilds';

// Com filhos
// Sem eventos
// Especialização do componente Hero
// Suporte a header,body and footer
// Deve ser informar como filhos em sequência header,body e footer
// Caso seja informado 1 filho : Será montado apenas o header
// Caso sejam informados 2 filhos : Será montado header e body
// Caso sejam informados 3 filhos: Será montado header,body e footer



const HeroThreeParts = (props)=>{
    var childrens =  React.Children.map(props.children, (child, i) => {return child});
    return assemble(childrens,props);
}

function assemble(childrens,props){
  var validChildrensNumber = (childrens.length <= 3) ? true : false;;
  var code = <></>;
  if(validChildrensNumber){
    code = assembleThreePartsHero(props,childrens);
  }else{
    throw new InvalidNumberChilds(assembleDefaultMessage(childrens.length));
  }

  return code;
}
// recebe a props e junto com os filhos monta head,body e footer
function assembleThreePartsHero(props,childrens){
    var definition =  classnames(getClassName("hero "+ props.definition,"Hero"));
    return (
        <section className={definition}>
            <div class="hero-head">
                {childrens[0] != undefined && childrens[0]}
            </div>
            
            <div className="hero-body">
                {childrens[1] != undefined && childrens[1]}
            </div>

            <div class="hero-foot">
            {childrens[2] != undefined && childrens[2]}
            </div>

        </section>);
}
HeroThreeParts.propTypes ={
    definition: PropTypes.string
}

export default HeroThreeParts;