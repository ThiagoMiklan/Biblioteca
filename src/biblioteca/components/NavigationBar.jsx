import React from 'react';
import PropTypes from 'prop-types';
import classname from 'classnames';
import getClassName from '../../tools/getClassName';
import { renderToString } from 'react-dom/server';

class NavigationBar extends React.Component {



    render() {
       // var startItens = getStartItens(this.props.itens);
        //var endItens  =  getEndItens(this.props.itens);
        return (
            <nav class="navbar">
                <div class="navbar-brand">
                    <a class="navbar-item" href={this.props.link_brand}>
                        <img src={this.props.src_brand} alt={this.props.alt} width={this.props.width_brand} height={this.props.height_brand}></img>
                    </a>
                </div>
                <div className="navbar-menu">
                    <div className="navbar-start">
                        {assemble(this.props.itens_start)}
                    </div>
                    <div className="navbar-end">
                        {assemble(this.props.itens_end)}
                    </div>
                </div>
            </nav>

        );
    }

}
// Método passado para função filter da propriedade itens para que filter os itens que irão no start da barra
function filterStartItens(item) {
    var start_condition = (item["start"] == true && (item["end"] == false || item["end"] == undefined)) ;
    // default condition para casa não seja informado nem start e nem end, acontecendo isso é inserido no start
    var default_condition = (item["start"] == undefined || item["end"] == undefined);

    return start_condition || default_condition; 
}

//Método passado para função filter da propriedade itens para que filter os itens que irão no start da barra
function filterEndItens(item) {
    return item["end"] == true && (item["start"] == false || item["start"] == undefined);
}

function getStartItens(itens) {
    return itens.filter(filterStartItens);
}

function getEndItens(itens) {
    return itens.filter(filterEndItens);
}

function assemble(itens) {
    var itens_code = "";
    
    if (Array.isArray(itens)) {
        // no caso é necessário adicionar primeiro um item para que o dropdown possa funcionar
        if(itens.length > 0){
            itens_code = assembleItens(itens);
        }
    } else {
        itens_code = assembleItem(itens, itens["definition"]);
    }

    var itens_code_string = renderToString(itens_code);
    return itens_code;
}

function assembleItens(itens) {

    var itens_code =
        <div className="navbar-item has-dropdown is-hoverable">
            <a className="navbar-link">
                More
            </a>
            <div className="navbar-dropdown">
                {itens.map(item => assembleItem(item, ""))}
        </div>
        </div>;
    return itens_code;

}

function assembleItem(item, definition) {
    var className = "navbar-item " + classname(getClassName(definition, "NavigationBar"));
    var item_code = <a className={className}>{item["value"]}</a>;
    
    return item_code;
}

NavigationBar.propTypes = {
    src_brand: PropTypes.string,
    link_brand: PropTypes.string,
    width_brand: PropTypes.number,
    height_brand: PropTypes.number,
    alt_brand: PropTypes.string,
    itens_start : PropTypes.array,
    itens_end : PropTypes.array
}

export default NavigationBar;

