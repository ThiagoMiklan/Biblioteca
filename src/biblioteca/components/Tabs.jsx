import React from 'react';
import PropTypes from 'prop-types';
import classname from 'classnames';
import getClassName from '../../tools/getClassName';
import assembleListTabs  from '../../tools/assembleListTabs';
import Icon from '../elements/Icon';

const Tabs = (props)=>{
var definition = "tabs "+ classname(getClassName(props.definition,"Tabs"));

    return (<div className={definition}>
            {assembleListTabs(props.itens)}
            </div>
  );
}

export default Tabs;