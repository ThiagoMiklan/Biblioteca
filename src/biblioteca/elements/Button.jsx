import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import button_props from '../../props/ButtonProps.js'
import getClassNames from '../../tools/getClassName';

/*
Observações: 

    1 - Caso exista a necessidade de adicionar um ícone ao componente Button, é necessário seguir
             a seguinte estrutura:

            <Button definition="info" >
                <Icon icon_name="fas fa-check" [right={true} OR empty prop right]>
                    Nome do botão
                </Icon>
            </Button >

            Sendo necessário informar o componente Icon como filho do componente Button
            e o nome do botão deve ser informado no Icon e não no Button como em outras
            situações onde usa-se o componente isolado.
*/


class Button extends React.Component {

    constructor(props){
        super(props);
        this.state  = {definition:props.definition};
    }

    render() {
        var definition = getDefinition(this.props.delete, this.state.definition);

        return (
            <button disabled = {this.props.disabled == true? true: false } className={definition} onClick={getClickEvent(this.props.onClick).bind(this)}>
                {this.props.children}
         </button>
         );
    }
}

/* Se propriedade isDelete é verdadeira, então
    retorna somente o className "definition", não
    envolvendo outras classes do Bulma.
*/
function getDefinition(isDelete,definition){
    if(isDelete){
       return "delete";
    }else{
        return classnames(getClassNames("button "+ definition,"Button"));
    }
}
// caso seja informado onClick ele retorna a função retornada, senão retorna uma função vazia. Método criado para evitar problemas com undefined
function getClickEvent(onClick){

    if(typeof onClick == "function"){
        return onClick;
    }else{
        return ()=> {};
    }
    
}



Button.propTypes = {
    // definição CSS Bulma
    definition: PropTypes.string,
    // Método Onclick disponível
    onClick: PropTypes.func,
    // Caso exista um ícone, informar através da prop icon, passando o componente Icon
    icon: PropTypes.element,
    // Caso o botão seja do tipo delete
    delete: PropTypes.bool
};


export default Button;