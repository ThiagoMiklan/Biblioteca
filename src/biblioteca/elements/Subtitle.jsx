import {React,useState} from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName'
import {renderToString} from 'react-dom/server'


const Subtitle = (props)=>{
    
    return assembleTitle(props);
}

function assembleTitle(props){
     var map = mapFunctionsTitle();
     var definition =  "subtitle "+ classnames(getClassName(props.definition,"Title"));
     var title = map.get(props.definition)(definition,props.value);
     var code_title =  "";
     if(title != undefined){
        code_title =  title;
     }
    return code_title;
}

function mapFunctionsTitle(){
    var map = new Map();
    map.set("1",assembleH1);
    map.set("2",assembleH2);
    map.set("3",assembleH3);
    map.set("4",assembleH4);
    map.set("5",assembleH5);
    map.set("6",assembleH6);
    return map;
}

var assembleH1 = (definition,value) =>{
return (<h1 className={definition}>{value}</h1>);
}

var assembleH2 = (definition,value)=>{
    return (<h2  className={definition}>{value}</h2>);
}

var assembleH3 = (definition,value) =>{
    return (<h3  className={definition}>{value}</h3>);
}

var  assembleH4 = (definition,value) =>{
    return (<h4 className={definition}>{value}</h4>);
}

var assembleH5= (definition,value)=>{
    return (<h5 className={definition}>{value}</h5>);
}

var assembleH6 = (definition,value) =>{
    return (<h6  className={definition}>{value}</h6>);
}



export default Subtitle;