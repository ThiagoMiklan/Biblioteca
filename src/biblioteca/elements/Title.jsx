import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName'
import {renderToString} from 'react-dom/server'


const Title = (props)=>{
    return assembleTitle(props)
};
     
function assembleTitle(props){
     var map = mapFunctionsTitle();
     var definition =  "title "+ classnames(getClassName(props.definition,"Title"));
     var title = map.get(props.definition)(definition,props.children,props.p);
     var code_title =  "";
     if(title != undefined){
        code_title =  title;
     }
    return code_title;
}

function mapFunctionsTitle(){
    var map = new Map();
    map.set("1",assemble1);
    map.set("2",assemble2);
    map.set("3",assemble3);
    map.set("4",assemble4);
    map.set("5",assemble5);
    map.set("6",assemble6);
    return map;
}

// p ==  paragráfo, se a tag a ser utilizada é parágrafo ou tag de título
var assemble1 = (definition,value,p) =>{
    var code = "";
    if(p == true){
        code = <p className={definition}>{value}</p>; 
    }else{
        code = <h1 className={definition}>{value}</h1>; 
    }
    return code;
}

var assemble2 = (definition,value,p)=>{
    var code = "";
    if(p == true){
        code = <p className={definition}>{value}</p>; 
    }else{
        code= <h2 className={definition}>{value}</h2>; 
    }
    return code;
}

var assemble3 = (definition,value,p) =>{
    var code= "";
    if(p == true){
        code = <p className={definition}>{value}</p>; 
    }else{
        code = <h3 className={definition}>{value}</h3>; 
    }
    return code;
}

var  assemble4 = (definition,value,p) =>{
    var code= "";
    if(p == true){
        code = <p className={definition}>{value}</p>; 
    }else{
        code = <h4 className={definition}>{value}</h4>; 
    }
    return code;
}

var assemble5= (definition,value,p)=>{
    var code = "";
    if(p == true){
        code = <p className={definition}>{value}</p>; 
    }else{
        code = <h5 className={definition}>{value}</h5>; 
    }
    return code;
}

var assemble6 = (definition,value,p) =>{
    var code = "";
    if(p == true){
        code = <p className={definition}>{value}</p>; 
    }else{
        code= <h6 className={definition}>{value}</h6>; 
    }
    return code;
}


Title.propTypes ={
    value: PropTypes.string.isRequired,
    definition: PropTypes.string.isRequired
}

export default Title;