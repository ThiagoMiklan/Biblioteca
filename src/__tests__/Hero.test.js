import React from 'react';
import ReactDOM from 'react-dom';
import Hero from '../biblioteca/Hero';
import Button from '../biblioteca/Button';
import NavigationBar from '../biblioteca/NavigationBar';
import HeroThreeParts from '../biblioteca/HeroThreeParts';
import Title from '../biblioteca/Title';
import Subtitle from '../biblioteca/Subtitle';
import Tabs from '../biblioteca/Tabs';
import renderer from 'react-test-renderer';


var itens_navbar_start = [{ value: "Home" }, { value: "Documentation" }]
var itens_navbar_end = [{ value: <Button definition="dark">Github</Button> }, { value: <Button definition="link">Download</Button> }]
var itens_tabs = [{ value: "Modifiers" }, { value: "Grid" }, { value: "Elements" }, { value: "Components" }, { value: "Layout" }]


test('hero colors', () => {
    const colors = renderer.create(
        <>
            <Hero title="Hero Title" subtitle="Hero Subtitle" />
            <Hero definition="primary" title="Primary Title" subtitle="Primary Subtitle" />
            <Hero definition="info" title="Info Title" subtitle="Info Subtitle" />
            <Hero definition="success" title="Success Title" subtitle="Success Subtitle" />
            <Hero definition="warning" title="Warning Title" subtitle="Warning Subtitle" />
            <Hero definition="danger" title="Danger Title" subtitle="Danger Subtitle" />
            <Hero definition="light" title="Light Title" subtitle="Light Subtitle" />
            <Hero definition="dark" title="Dark Title" subtitle="Dark Subtitle" />
        </>
    ).toJSON();

    expect(colors).toMatchSnapshot();
})

test('hero bold', () => {
    const bold = renderer.create(
        <>
            <Hero definition="primary bold" title="Primary Title" subtitle="Primary Subtitle" />
            <Hero definition="info bold" title="Info Title" subtitle="Info Subtitle" />
            <Hero definition="success bold" title="Success Title" subtitle="Success Subtitle" />
            <Hero definition="warning bold" title="Warning Title" subtitle="Warning Subtitle" />
            <Hero definition="danger bold" title="Danger Title" subtitle="Danger Subtitle" />
            <Hero definition="light bold" title="Light Title" subtitle="Light Subtitle" />
            <Hero definition="dark bold" title="Dark Title" subtitle="Dark Subtitle" />
        </>
    ).toJSON();

    expect(bold).toMatchSnapshot();
})

test('hero merge', () => {
    const merge = renderer.create(
        <>
        <Hero definition="primary medium" title="Primary Title" subtitle="Primary Subtitle" />
        <Hero definition="info large" title="Info Title" subtitle="Info Subtitle" />
        <Hero definition="success large" title="Success Title" subtitle="Success Subtitle" />
        </>
    ).toJSON();

    expect(merge).toMatchSnapshot();
})

test('navigation bar',()=>{
    const nav = renderer.create(
        <>
        <NavigationBar itens_start={itens_navbar_start} itens_end={itens_navbar_end} />
        <Hero definition="info with-navbar" title="With navbar" />
        </>
    ).toJSON();

    expect(nav).toMatchSnapshot();
})
