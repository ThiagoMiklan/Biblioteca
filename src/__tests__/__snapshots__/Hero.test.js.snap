// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hero bold 1`] = `
Array [
  <section
    className="hero is-primary is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Primary Title
        </h1>
        <h2
          className="subtitle"
        >
          Primary Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-info is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Info Title
        </h1>
        <h2
          className="subtitle"
        >
          Info Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-success  is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Success Title
        </h1>
        <h2
          className="subtitle"
        >
          Success Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-warning is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Warning Title
        </h1>
        <h2
          className="subtitle"
        >
          Warning Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-danger is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Danger Title
        </h1>
        <h2
          className="subtitle"
        >
          Danger Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-light is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Light Title
        </h1>
        <h2
          className="subtitle"
        >
          Light Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-dark is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Dark Title
        </h1>
        <h2
          className="subtitle"
        >
          Dark Subtitle
        </h2>
      </div>
    </div>
  </section>,
]
`;

exports[`hero colors 1`] = `
Array [
  <section
    className="hero"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Hero Title
        </h1>
        <h2
          className="subtitle"
        >
          Hero Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-primary"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Primary Title
        </h1>
        <h2
          className="subtitle"
        >
          Primary Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-info"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Info Title
        </h1>
        <h2
          className="subtitle"
        >
          Info Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-success"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Success Title
        </h1>
        <h2
          className="subtitle"
        >
          Success Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-warning"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Warning Title
        </h1>
        <h2
          className="subtitle"
        >
          Warning Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-danger"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Danger Title
        </h1>
        <h2
          className="subtitle"
        >
          Danger Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-light"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Light Title
        </h1>
        <h2
          className="subtitle"
        >
          Light Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-dark"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Dark Title
        </h1>
        <h2
          className="subtitle"
        >
          Dark Subtitle
        </h2>
      </div>
    </div>
  </section>,
]
`;

exports[`hero merge 1`] = `
Array [
  <section
    className="hero is-primary is-medium"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Primary Title
        </h1>
        <h2
          className="subtitle"
        >
          Primary Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-info is-large"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Info Title
        </h1>
        <h2
          className="subtitle"
        >
          Info Subtitle
        </h2>
      </div>
    </div>
  </section>,
  <section
    className="hero is-success is-large"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Success Title
        </h1>
        <h2
          className="subtitle"
        >
          Success Subtitle
        </h2>
      </div>
    </div>
  </section>,
]
`;

exports[`navigation bar 1`] = `
Array [
  <nav
    className="navbar"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
      >
        <img />
      </a>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-start"
      >
        <a
          className="navbar-item undefined"
        >
          Home
        </a>
        <a
          className="navbar-item undefined"
        >
          Documentation
        </a>
      </div>
      <div
        className="navbar-end"
      >
        <a
          className="navbar-item undefined"
        >
          <button
            className="button is-dark"
            disabled={false}
          >
            Github
          </button>
        </a>
        <a
          className="navbar-item undefined"
        >
          <button
            className="button is-link"
            disabled={false}
          >
            Download
          </button>
        </a>
      </div>
    </div>
  </nav>,
  <section
    className="hero is-link is-fullheight-with-navbar"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          With navbar
        </h1>
        <h2
          className="subtitle"
        />
      </div>
    </div>
  </section>,
]
`;

exports[`three parts 1`] = `
<section
  className="hero is-fullheight"
>
  <div
    class="hero-head"
  >
    <nav
      className="navbar"
    >
      <div
        className="navbar-brand"
      >
        <a
          className="navbar-item"
        >
          <img />
        </a>
      </div>
      <div
        className="navbar-menu"
      >
        <div
          className="navbar-start"
        >
          <a
            className="navbar-item undefined"
          >
            Home
          </a>
          <a
            className="navbar-item undefined"
          >
            Documentation
          </a>
        </div>
        <div
          className="navbar-end"
        >
          <a
            className="navbar-item undefined"
          >
            <button
              className="button is-dark"
              disabled={false}
            >
              Github
            </button>
          </a>
          <a
            className="navbar-item undefined"
          >
            <button
              className="button is-link"
              disabled={false}
            >
              Download
            </button>
          </a>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <h1
        className="title is-1"
      >
        Title
      </h1>
      <h2
        className="subtitle is-2"
      >
        Subtitle
      </h2>
    </div>
  </div>
  <div
    class="hero-foot"
  >
    <div
      className="tabs"
    >
      <ul>
        <li>
          <a>
            Modifiers
          </a>
        </li>
        <li>
          <a>
            Grid
          </a>
        </li>
        <li>
          <a>
            Elements
          </a>
        </li>
        <li>
          <a>
            Components
          </a>
        </li>
        <li>
          <a>
            Layout
          </a>
        </li>
      </ul>
    </div>
  </div>
</section>
`;
